provider-service/base/pkg/streams/sources/pub_sub_source.go

```diff
@@ -0,0 +1,120 @@
+package sources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/go-logr/logr"
+	"github.com/sky-uk/kfp-operator/argo/common"
+	. "github.com/sky-uk/kfp-operator/provider-service/base/pkg"
+)
+
+type PubSubSource struct {
+	RunsSubscription *pubsub.Subscription
+	Logger           logr.Logger
```

Comment:
should we be passing the logger around, I think this would be better passing the context and extracting the logger where its used instead

Reply:
yea passing the context around is better since zap is well integrated with context

Reply:
done